### PR TITLE
Use pure numpy arrays in power_output calculation (~x10 speedup for large datasets)

### PR DESCRIPTION
**Pull request by kyri-petrou**
Fixes # .

Changes proposed in this pull request:
- Use of pure numpy arrays in the power output calculation in power_output.power_curve_density_correction(). On a ~220,000 timestamps dataset, the computation time was reduced from 50 seconds to ~3 seconds.

### CODE REVIEW

File: windpowerlib/power_output.py [hunk 1]
```diff
@@ -247,24 +247,12 @@ def power_curve_density_correction(
...
-            )
-        )
-        for i in range(len(wind_speed))
-    ]
+    power_output = _get_power_output(
+        wind_speed,
+        power_curve_wind_speeds.to_numpy(),
+        density.to_numpy(),
+        power_curve_values.to_numpy(),
+    )
```
Comment (uvchik) [inline windpowerlib/power_output.py:255]: The problem is that we also allow numpy-arrays as an input, in that case `to_numpy` will not work.

I think `numpy.array()` should work for Series and Arrays.

```diff
- power_curve_wind_speeds.to_numpy(),
+ numpy.array(power_curve_wind_speeds)
```

**Review by uvchik — COMMENTED** [review]
Thank you, @kyri-petrou for your contribution and welcome to our community :smile: 

My comment could help to fix the failing tests.

File: windpowerlib/power_output.py (hunk 1 above)
Comment (kyri-petrou) [reply windpowerlib/power_output.py]: Ah thanks for the comments! I couldn't see what was failing with Travis, only that an error occured.

Hopefully the new commit fixes it

**Comment by uvchik** [comment]
I divided the very long test method into single methods to make it easier to find the failing test. We should add a description to each test to make it clear what they do, but this could be a new PR.

Furthermore, I fixed the last failing test. This test was kind of strange, so in contrast to your fix above this failing test was not your fault.

Thank you for speeding up the windpowerlib!

**Review by uvchik — APPROVED** [review]

**Pull request merged**